### PR TITLE
Refactor/store

### PR DESCRIPTION
**Pull request by ChickenStorm**

### CODE REVIEW

File: game/map/system.gd [hunk 1]
```diff
@@ -10,10 +10,12 @@ const _SCALE_FACTOR_ON_HIGHLIGHT = 1.5
 const _SCALE_CHANGE_FACTOR = 5.0
 const SYSTEM_FLEET_PIN_SCENE = preload("res://game/map/system_fleet_pin.tscn")
 const SYSTEM_BUILDING_PIN_SCENE = preload("res://game/map/system_building_pin.tscn")
+const ASSETS = preload("res://resources/assets.tres")
 
 export(float) var scale_ratio = 1.0 setget set_scale_ratio
 
-var system = null
+var game_data : GameData = Store.game_data
```
Comment (Kern046) [inline game/map/system.gd]:
```suggestion
var _game_data : GameData = Store.game_data
```

File: game/map/system.gd [hunk 2]
```diff
@@ -96,38 +105,39 @@ func update_ship_pin():
 
 func _update_ship_pin_number(number):
 	for node in building_pins.get_children():
-		if node.building.kind == "shipyard":
-			node.blinking = not(number == 0)
+		if node.building.kind.kind == "shipyard":
```
Comment (Kern046) [inline game/map/system.gd:108]: Sure of this ?

File: hud/system/buildings/contruction/construction_building_item.gd [hunk 3]
```diff
@@ -3,8 +3,11 @@ extends MarginContainer
 
 signal building_contructing(building)
 
-export(String) var building_type = null setget set_building_type
+const ASSETS = preload("res://resources/assets.tres")
 
+export(Resource) var building_type = null setget set_building_type # ressource
+
+var game_data : GameData = Store.game_data
```
Comment (Kern046) [inline hud/system/buildings/contruction/construction_building_item.gd]:
```suggestion
var _game_data : GameData = Store.game_data
```

File: hud/system/buildings/hangar/ship_type_build.gd [hunk 4]
```diff
@@ -21,24 +24,24 @@ func _ready():
...
 	_update_max_values()
 
 
-func _on_text_entered(text = null):
+func _on_text_entered(_text = null):
 	_on_pressed_build()
 
 
-func _on_wallet_update(ammount):
+func _on_wallet_update(_ammount):
```
Comment (Kern046) [inline hud/system/buildings/hangar/ship_type_build.gd]: _amount

File: matchmaking/lobby/faction_column.gd [hunk 5]
```diff
@@ -21,13 +22,21 @@ func _ready():
...
+	for node in container.get_children():
+		if node is PlayerInfoFactionColumn and node.player.id == player.id:
+			has_player = true
+			break
+	if not has_player:
+		var node = PLAYER_INFO_FACTION_COLUMN.instance()
+		node.player = player
+		container.add_child(node)
+		return UPDATE_PLAYER_STATE.PLAYER_ADDED
+	else:
```
Comment (Kern046) [inline matchmaking/lobby/faction_column.gd]: ``else`` statement is not useful if your ``if`` has a return statement.

For proper code, I would have inverted the condition, so the ``if`` has just the ``return`` statement currently contained in the ``else``, and the rest of the code has one less indentation level

File: matchmaking/lobby/game_settings.gd [hunk 6]
```diff
@@ -27,37 +24,40 @@ func _update_enable_state():
 
 
 func _on_speed_selected(id):
-	_pacth_game_seting("game_speed", SPEED[id])
+	Store.lobby.option.game_speed = LobbyOption.SPEED[id]
+	_pacth_game_seting(Store.lobby.option)
```
Comment (Kern046) [inline matchmaking/lobby/game_settings.gd]:
```suggestion
	_patch_game_setting(Store.lobby.option)
```

File: resources/assets/2d/picto/building/building.gd [hunk 7]
```diff
@@ -0,0 +1,25 @@
+tool
+extends BuildingModelRemote
+class_name KalaxiaBuilding
+
+const NAME_OF_RESSOURCE = {
```
Comment (Kern046) [inline resources/assets/2d/picto/building/building.gd]:
```suggestion
const RESOURCE_NAME = {
```

File: resources/assets/2d/picto/ships/stat/ship_model.gd [hunk 8]
```diff
@@ -0,0 +1,16 @@
+extends Node
```
Comment (Kern046) [inline resources/assets/2d/picto/ships/stat/ship_model.gd]: Is this file useful ?

File: resources/definitions/remote_resources/cached_resource.gd [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+func load_constant():
+	constants = ConstanteRemoteResource.new()
+	constants.load_remote()
+	constants.connect("loaded", self, "_on_constants_loaded")
+	constants.connect("error_loading", self, "_on_constants_error_loading")
+
+
+func has_all_data():
+	return constants.has_all_data() and has_ships_model() \
+	and has_factions() and has_building()
```
Comment (Kern046) [inline resources/definitions/remote_resources/cached_resource.gd]:
```suggestion
	    and has_factions() and has_building()
```

File: resources/definitions/store/building.gd [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+
+func _init(dict = null).(dict):
+	pass
+
+
+func load_dict(dict):
+	if dict == null:
+		return
+	.load_dict(dict)
+	if dict.kind is KalaxiaBuilding:
```
Comment (Kern046) [inline resources/definitions/store/building.gd]: A ternary would be more appropriate

File: resources/definitions/store/lobby.gd [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+	if dict == null:
+		return
+	.load_dict(dict)
+	if (not dict is Dictionary or dict.has("players")) and dict.players != null:
+		players.clear()
+		for player in dict.players:
+			if player is String:
+				players[player.id] = Player.new({"id" : player})
+			else:
+				players[player.id] = Player.new(player)
```
Comment (Kern046) [inline resources/definitions/store/lobby.gd]:
```suggestion
			players[player.id] = Player.new({"id" : player} if player is String else player)
```

File: resources/definitions/store/lobby.gd [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+	if not dict is Dictionary or dict.has("owner"):
+		if dict.owner is String:
+			owner = Player.new({"id" : dict.owner})
+		else:
+			owner = Player.new(dict.owner)
+		players[owner.id] = owner # relink owner to the array of player
```
Comment (Kern046) [inline resources/definitions/store/lobby.gd]: Not sure of this due to your comment but worth the try :D !
```suggestion
		players[owner.id] = Player.new({"id" : dict.owner} if dict.owner is String else dict.owner) # relink owner to the array of player
```

File: resources/definitions/store/lobby.gd [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+	if not pid is String and not (pid is Dictionary and pid.has("id")):
+		return null
+	for p in players.values():
+		if (pid is String and p.id == pid) or (pid is Dictionary and pid.id == p.id):
+			return p
+	return null
+
+
+func remove_player_lobby(player_id : String) -> bool:
+	var has_ereased = players.erase(player_id)
```
Comment (Kern046) [inline resources/definitions/store/lobby.gd]:
```suggestion
	var is_erased = players.erase(player_id)
```

File: resources/definitions/store/ship_group.gd [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+func _init(dict = null).(dict):
+	pass
+
+
+func load_dict(dict : Dictionary) -> void:
+	if dict == null:
+		return
+	.load_dict(dict)
+	if not dict is Dictionary or dict.has("category"):
+		if dict.category is KalaxiaShipModel:
```
Comment (Kern046) [inline resources/definitions/store/ship_group.gd]: Ternary

**Review by Kern046 — APPROVED** [review]

File: resources/assets/2d/picto/building/building.gd (hunk 7 above)
Comment (ChickenStorm) [reply resources/assets/2d/picto/building/building.gd]: corrected on #75

File: resources/definitions/store/lobby.gd (hunk 12 above)
Comment (ChickenStorm) [reply resources/definitions/store/lobby.gd]: `owner` has to be set, I used a ternary to set `owner` first

**Pull request merged**